### PR TITLE
AlmaLinux/build-system#24: Add mechanism of generating community PGP keys

### PR DESCRIPTION
**Pull request by soksanichenko**
- The DB migration for new table `gen_key_tasks` and the new fields `is_community` and `product_id` for table `sign_keys`.
- Three new endpoints:
  - `/{product_id}/{platform_name}/gen-sign-key/`: accepts a POST request to generate a new gen key task. It's called from frontend side.
  - `/sign-tasks/community/get_gen_sign_key_task/`: accepts a POST request to get a first available gent key task. It's called by the sign node.
  - `/sign-tasks/community/{gen_key_task_id}/complete/`: accepts a POST request to complete a gen key task. It's called by the sign node.
- Some refactoring
- Create a repo for a product's sign key when a product is being created
- The separate statuses enum for a gen key task

### CODE REVIEW

**Review by zklevsha — APPROVED** [review]

File: tests/test_api/test_products.py [hunk 1]
```diff
@@ -25,6 +29,20 @@ async def test_product_create(self, product_create_payload):
         )
         assert response.status_code == self.status_codes.HTTP_200_OK, message
 
+    async def test_product_remove(
```
Comment (anfimovdm) [inline tests/test_api/test_products.py:32]: we already have the similar test, it's `test_user_product_remove`

**Review by anfimovdm — CHANGES_REQUESTED** [review]

File: alws/crud/sign_task.py [hunk 2]
```diff
@@ -157,6 +189,28 @@ async def create_sign_task(
...
+                .selectinload(models.Product.owner),
+            )
+        )
+        gen_key_task = gen_key_tasks.scalars().first()
+        if gen_key_task:
+            await db.execute(
+                update(models.GenKeyTask)
+                .where(models.GenKeyTask.id == gen_key_task.id)
+                .values(status=GenKeyStatus.IN_PROGRESS)
+            )
```
[the remark this answers is not in the document]
Comment (soksanichenko) [reply alws/crud/sign_task.py:210]: It's not atomic

File: Dockerfile [hunk 3]
```diff
@@ -8,8 +8,8 @@ RUN mkdir -p /code && \
     yum clean all
 RUN curl https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh -o wait_for_it.sh && chmod +x wait_for_it.sh
 COPY requirements.txt /tmp/requirements.txt
-RUN cd /code && virtualenv -p python3.9 --system-site-packages env && source env/bin/activate \
-    && pip3 install -r /tmp/requirements.txt --no-cache-dir
+RUN cd /code && virtualenv -p python3.9 --system-site-packages env && source env/bin/activate && pip3 install -U pip
+RUN cd /code && source env/bin/activate && pip3 install -r /tmp/requirements.txt --no-cache-dir
```
[the remark this answers is not in the document]
Comment (soksanichenko) [reply Dockerfile:12]: Joined into the one layer

File: alws/crud/sign_task.py [hunk 4]
```diff
@@ -157,6 +189,28 @@ async def create_sign_task(
...
+    return gen_key_task
```
[the remark this answers is not in the document]
Comment (soksanichenko) [reply alws/crud/sign_task.py:211]: Fixed

File: alws/crud/sign_task.py [hunk 5]
```diff
@@ -259,6 +313,95 @@ async def get_sign_task(
...
+        task_status = GenKeyStatus.FAILED
+        error_message = payload.error_message
+    await db.execute(
+        update(models.GenKeyTask)
+        .where(models.GenKeyTask.id == gen_key_task_id)
+        .values(
+            status=task_status,
+            error_message=error_message,
+        )
+    )
```
[the remark this answers is not in the document]
Comment (soksanichenko) [reply alws/crud/sign_task.py:342]: it's not atomic

File: alws/routers/sign_task.py [hunk 6]
```diff
@@ -119,3 +119,38 @@ async def iter_sync_sign_tasks(
...
+async def get_avaiable_gen_key_task(db: database.Session = Depends(get_db)):
+    gen_key_task = await sign_task.get_available_gen_key_task(db)
+    if gen_key_task:
+        return {
+            'id': gen_key_task.id,
+            'product_name': gen_key_task.product.name,
+            'user_name': gen_key_task.product.owner.username,
+            'user_email': gen_key_task.product.owner.email,
+        }
+    else:
```
[the remark this answers is not in the document]
Comment (soksanichenko) [reply alws/routers/sign_task.py:137]: Why?

File: tests/test_api/test_products.py (hunk 1 above)
Comment (soksanichenko) [reply tests/test_api/test_products.py:32]: That's not mine

**Pull request merged**